adapter/grpc/client.go

```diff
@@ -0,0 +1,72 @@
+package grpc
+
+import (
+	"context"
+
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+	"google.golang.org/grpc"
+)
+
+// UnaryClientIntercept returns new grpc.UnaryClientInterceptor instance
+func UnaryClientIntercept(opts ...Option) grpc.UnaryClientInterceptor {
+	options := evaluateOptions(opts)
+	return func(
+		ctx context.Context,
+		method string,
+		req, reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		// method as resource name by default
+		resourceName := method
+		if options.unaryClientResourceExtract != nil {
+			resourceName = options.unaryClientResourceExtract(ctx, method, req, cc)
+		}
+
+		entry, err := sentinel.Entry(
+			resourceName,
+			sentinel.WithResourceType(base.ResTypeRPC),
+			sentinel.WithTrafficType(base.Outbound),
+		)
+		if err != nil {
+			return err
+		}
+		defer entry.Exit()
+
+		return invoker(ctx, method, req, reply, cc, opts...)
```

Comment:
Actually here we could record the error (if present) for later use (e.g. circuit breaking). We may leverage `sentinel.TraceErrorToEntry(entry, err)` to do this.

Reply:
We can handle errors here by injecting a new interceptor after `UnaryClientIntercept`, if you really care about the `blockerror`(like circuit breaking).

Also, we can add a `blockFallback` option to handle `blockerror` just like i do in gin adapter. But a little redundant is that the type of blockFallback is grpc.UnaryClientInterceptor.